python/res/fm/ecl/ecl_run.py

```diff
@@ -164,77 +171,77 @@ def initMPI(self):
                 fileH.write("%s\n" % host)
 
 
-
-    def initRedirect(self):
+    def execSerialEclipse(self):
         stdout_file = "%s.LOG" % self.base_name
         stderr_file = "%s.stderr" % self.base_name
+        with open(stdout_file, "w") as stdout, open(stderr_file, "w") as stderr:
+            process = subprocess.Popen(
+                [self.sim.executable, self.base_name],
+                env=self.sim.env,
+                stdout=stdout,
+                stderr=stderr
+            )
+            process.communicate()
+            return process.wait()
 
-        # Redirecting stdout / stderr
-        fd_stdout = os.open(stdout_file , os.O_WRONLY | os.O_TRUNC | os.O_CREAT , 0o644);
-        fd_stderr = os.open(stderr_file , os.O_WRONLY | os.O_TRUNC | os.O_CREAT , 0o644);
 
-        os.dup2(fd_stdout , 1)
-        os.dup2(fd_stderr , 2)
+    def execParallellEclipse(self):
+        self.initMPI( )
+        stdout_file = "%s.LOG" % self.base_name
+        stderr_file = "%s.stderr" % self.base_name
+        with open(stdout_file, "w") as stdout, open(stderr_file, "w") as stderr:
+            process = subprocess.Popen(
+                [self.sim.mpirun,
+                 "-machinefile",
+                 self.machine_file,
+                 "-np",
+                 "%s" % self.num_cpu,
+                 self.sim.executable,
+                 self.base_name],
+                env=self.sim.env,
+                stdout=stdout,
+                stderr=stderr)
+            process.communicate()
+            return process.wait()
 
-        os.close(fd_stdout)
-        os.close(fd_stderr)
 
+    def execEclipse(self):
+        with temp_dir_change():
 
+            os.chdir(self.run_path)
 
-    def execSerialEclipse(self):
-        os.execv( self.sim.executable , [ self.sim.executable , self.base_name ])
+            if not os.path.exists( self.data_file ):
+                raise IOError("Can not find data_file:%s" % self.data_file )
 
+            if not os.access( self.data_file , os.R_OK ):
+                raise OSError("Can not read data file:%s" % self.data_file )
 
-    def execParallellEclipse(self):
-        self.initMPI( )
-        os.execv( self.sim.mpirun , [self.sim.mpirun , "-machinefile" , self.machine_file , "-np" , "%s" % self.num_cpu , self.sim.executable , self.base_name ] )
-
+            if self.num_cpu == 1:
+                return_code = self.execSerialEclipse()
```

Comment:
With the context manager you can go back to returning directly here if you prefer ;)